src/tools/ScenarioMeasurement/Startup/GenericStartupParser.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Session;
+using Reporting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ScenarioMeasurement
+{
+    // [EventSource(Guid = "9bb228bd-1033-5cf0-1a56-c2dbbe0ebc86")]
+    // class PerfLabGenericEventSource : EventSource
+    // {
+    //     public static PerfLabGenericEventSource Log = new PerfLabGenericEventSource();
+    //     public void Startup() => WriteEvent(1);
+    // }
+
+    internal class GenericStartupParser : IParser
+    {
+        public void EnableKernelProvider(TraceEventSession kernel)
+        {
+            kernel.EnableKernelProvider((KernelTraceEventParser.Keywords)(KernelTraceEventParser.Keywords.Process | KernelTraceEventParser.Keywords.Thread | KernelTraceEventParser.Keywords.ContextSwitch));
+        }
+
+        public void EnableUserProviders(TraceEventSession user)
+        {
+            user.EnableProvider("PerfLabGenericEventSource");
```

Comment:
post-merge question: can we skip creating a user provider if it's not used anywhere?

Reply:
> Can we skip creating a user provider and just having an empty method for the interface?

Yes, you can choose not to create the session if you don't need it.  Though it's probably easiest from a code perspective to not worry about that and always create it.  The cost of having the session around is negligible.

>Is "PerfLabGenericEventSource" pre-defined somewhere?

ETW does not have a hard requirement that providers are registered anywhere.  The key is that if you have something that emits an event, then you probably need something that knows how to read it.  There are a few options around this - one of them is to register providers with the operating system.  For `EventSource` implementations, we generally don't do this - instead the `EventSource` writes its metadata into the trace and then the consuming application just knows how to pull the metadata out of the trace and use it.  In terms of this app, you should expect to see a class called PerfLabGenericEventSource somewhere in the apps that are being traced, as they'll need to emit an event that represents the end of start-up.

Reply:
Looks like  PerfLabGenericEventSource class is commented out

Reply:
Yup.  Not sure if that's intentional or not.  @billwert?

Reply:
It's intentional. The expectation is that apps/tests that want to use it will just copy it in. There's very little chance we will meaningfully change it in the future in a way that might break some measurement (we're more likely to just make a new one). I don't want to build a separate DLL because I don't want to introduce a module load into the path of logging events.